_spikes-poc/mssql-test/app.js

```diff
@@ -0,0 +1,74 @@
+const createError = require('http-errors');
+const express = require('express');
+const path = require('path');
+const cookieParser = require('cookie-parser');
+const morgan = require('morgan');
+const logger = require('./logger')
+const sqlService = require('./sql.service')
+
+function sleep(ms) {
+  return new Promise(resolve => {
+    setTimeout(resolve, ms)
+  })
+}
+
+;(async function () {
+  let connected = true
+  let backOff = 2 // milliseconds
+  let startTime
+  do {
+    startTime = new Date()
+    try {
+      logger.warn('Attempting to initialise the connection pool')
+      await sqlService.initPool()
+      connected = true
+    } catch (error) {
+      // await sleep(2000)
+      // process.exit(1)
+      await sqlService.drainPool()
+      sqlService.pool = null
```

Comment:
Is it going to assume there is still a connection pool if you don't set it to null I suppose?

Reply:
This block of code basically represents my expectation of how I thought the connection handling should work.    Draining and setting the pool to null were my increasingly desperate attempts to make the connection pool re-connect to the database.   However, If we bring the app up without the database, the connection pool is filled with closed connections, which are unusable, and they never re-connect.

In the admin app implementation I called process.exit() to terminate the app and allow the pm2 node manager to restart it after a delay of 30 seconds to allow the DB to come up.